phantom/agents.py

```diff
@@ -161,6 +161,17 @@ def reset(self) -> None:
         Can be extended by subclasses to provide additional functionality.
         """
 
+        if hasattr(self, "Supertype"):
```

Comment:
That means that here we must enforce the class to be called Supertype. Is there another way to check whether an agent has a supertype associated?

Reply:
Fixed. Now the agent will first check for an `Agent.supertype` supertype instance, if not will then check for an `Agent.Supertype` class to generate a default instance from, if not no supertype is set.